test-project/Assets/EditmodeTests/Utility/InterestTemplateTests.cs

```diff
@@ -0,0 +1,187 @@
+
+        [Test]
+        public void AddQueries_can_be_called_after_AddQueries()
+        {
+            Assert.DoesNotThrow(() => EmptyInterest
+                .AddQueries<Position.Component>(BasicQuery, BasicQuery)
+                .AddQueries<Metadata.Component>(BasicQuery, BasicQuery));
+        }
+
+        [Test]
+        public void AddQueries_can_be_called_with_a_single_query()
+        {
+            Assert.DoesNotThrow(() => EmptyInterest
+                .AddQueries<Position.Component>(BasicQuery));
+        }
+
+        [Test]
+        public void ReplaceQueries_clears_previous_query()
+        {
+            var initialQuery = BasicQuery;
+            var initialQueryRadius = initialQuery.AsComponentInterestQuery()
+                .Constraint.RelativeSphereConstraint.Value.Radius;
+
+            var differentBasicQuery = DifferentBasicQuery;
+            var differentQueryRadius = differentBasicQuery.AsComponentInterestQuery()
+                .Constraint.RelativeSphereConstraint.Value.Radius;
+
+            var interest = EmptyInterest
+                .AddQueries<Position.Component>(initialQuery)
+                .ReplaceQueries<Position.Component>(differentBasicQuery);
+
+            var queryExists = interest.AsComponentInterest().TryGetValue(Position.ComponentId, out var replacedQuery);
+            var replacedQueryRadius = replacedQuery.Queries[0].Constraint.RelativeSphereConstraint.Value.Radius;
+
+            Assert.True(queryExists);
+            Assert.AreEqual(1, replacedQuery.Queries.Count);
+            Assert.AreNotEqual(initialQueryRadius, replacedQueryRadius);
+            Assert.AreEqual(differentQueryRadius, replacedQueryRadius);
+        }
+
+        [Test]
+        public void ReplaceQueries_clears_previous_queries()
+        {
+            var interest = EmptyInterest
+                .AddQueries<Position.Component>(BasicQuery, BasicQuery, BasicQuery)
+                .ReplaceQueries<Position.Component>(DifferentBasicQuery);
+
+            var queryExists = interest.AsComponentInterest().TryGetValue(Position.ComponentId, out var replacedQuery);
+
+            Assert.True(queryExists);
+            Assert.AreEqual(1, replacedQuery.Queries.Count);
+        }
+
+        [Test]
+        public void ClearQueries_clears_a_single_list_of_queries()
+        {
+            var interest = EmptyInterest
+                .AddQueries<Metadata.Component>(BasicQuery)
+                .AddQueries<Persistence.Component>(BasicQuery)
+                .AddQueries<Position.Component>(BasicQuery)
+                .ClearQueries<Metadata.Component>();
+
+            Assert.AreEqual(2, interest.AsComponentInterest().Keys.Count);
+        }
+
+        [Test]
+        public void ClearQueries_can_be_used_with_empty_lists()
+        {
+            Assert.DoesNotThrow(() => EmptyInterest
+                .ClearQueries<Metadata.Component>());
+        }
+
+        [Test]
+        public void ClearAllQueries_clears_entire_dictionary()
```

Comment:
Test name references internal implementation (there are few more cases of this below, I won't repeat them all)